cylc/flow/option_parsers.py

```diff
@@ -51,20 +52,94 @@
-    metavar="CYCLE_POINT or OFFSET",
+APPEND = 'append'
+ARGS = 'args'
+KWARGS = 'kwargs'
+HELP = 'help'
+ACTION = 'action'
+STORE = 'store'
+STORE_CONST = 'store_const'
+STORE_TRUE = 'store_true'
+SOURCES = 'sources'
+DEFAULT = 'default'
+DEST = 'dest'
+METAVAR = 'metavar'
+CHOICES = 'choices'
+OPTS = 'opts'
+ALL = 'all'
+USEIF = 'useif'
+DOUBLEDASH = '--'
+
+
+class OptionSettings():
+    """Container for info about a command line option
+
+    Despite some similarities this is not to be confused with
+    optparse.Option: This a container for information which may or may
+    not be passed to optparse depending on the results of
+    cylc.flow.option_parsers(thismodule).combine_options_pair.
+    """
+
+    def __init__(self, argslist, sources=None, useif=None, **kwargs):
+        """Init function:
+
+        Args:
+            arglist: list of arguments for optparse.Option.
+            sources: set of CLI scripts which use this option.
+            useif: badge for use by Cylc optionparser.
+            **kwargs: kwargs for optparse.option.
+        """
+        self.kwargs: Dict[str, str] = {}
+        self.sources: set = sources if sources is not None else set()
+        self.useif: str = useif if useif is not None else ''
+
+        self.args: list[str] = argslist
+        for kwarg, value in kwargs.items():
+            self.kwargs.update({kwarg: value})
+
+    def __eq__(self, other):
+        """Args and Kwargs, but not other props equal."""
+        return (
+            self.kwargs == other.kwargs
+            and self.args == other.args
+        )
+
+    def __and__(self, other):
+        """Is there a set intersection between arguments."""
+        return list(set(self.args).intersection(set(other.args)))
+
+    def __sub__(self, other):
+        """Set difference on args."""
+        return list(set(self.args) - set(other.args))
+
+    def _in_list(self, others):
+        """CLI arguments for this option found in any of a list of
+        other options."""
+        return any([self & other for other in others])
+
+    def _update_sources(self, other):
+        """Update the sources from this and 1 other OptionSettings object"""
+        self.sources = {*self.sources, *other.sources}
+
+
+ICP_OPTION = OptionSettings(
+    ["--initial-cycle-point", "--icp"],
     help=(
-        "Set the initial cycle point. "
-        "Required if not defined in flow.cylc."
-        "\nMay be either an absolute point or an offset: See "
-        f"{SHORTLINK_TO_ICP_DOCS} (Cylc documentation link)."
+        "Set the initial cycle point."
+        " Required if not defined in flow.cylc."
+        "\nMay be either an absolute point or an offset: See"
+        f" {SHORTLINK_TO_ICP_DOCS} (Cylc documentation link)."
     ),
-    action="store",
-    dest="icp",
+    metavar="CYCLE_POINT or OFFSET",
+    action=STORE,
+    dest="icp"
 )
 
 
+icp_option = Option(
+    *ICP_OPTION.args, **ICP_OPTION.kwargs)  # type: ignore[arg-type]
```

Comment:
FYI you can provide args as kwargs and vice versa:

```python
>>> def foo(a, b=2):
...     print(a, b)
... 
>>> foo(1, b=3)
1 3
>>> foo(1, 3)
1 3
>>> foo(a=1, b=3)
1 3
>>> foo(b=3, a=1)
1 3
```

So you could potentially absorb the positional args into the kwargs for simplicity.